### PR TITLE
Add tenant host application with feature flag 

### PR DESCRIPTION
**Pull request by freva**
# TL;DR
This introduces a new infrastructure application (one that only exists in node-repository and orchestrator) for tenant hosts that will be enabled through a feature flag. The plan is:
* deploy a zone-application that does not have the `<nodes type="host">` element, this will deallocate all tenant hosts, making them `inactive`
* `InactiveExpirer` will expire (non-recursive move to `dirty`) all `host` nodes once run (interval is 2s in CD and 25m in main, the maintainers are evenly spaced across cfgs, so this means max 9m wait)
* Each tenant host will then transition to `ready`
* `InfrastructureProvisioner` will then prepare & activate the new tenant host app which will allocate all nodes of type `host`.

Next step will be to update `Zones` (`vespa/hosted#7846`) to specify to either:
1. Not generate any zone-app for the zone (if the zone doesn't have any `proxy` nodes)
2. Only generate the zone-app that allocates `proxy` nodes.

This is also closely related to #9643 which implements the new upgrade procedure of infrastructure in the controller.

### CODE REVIEW

**Comment by freva** [comment]
@hakonhall I don't know if there are other changes required in service-monitor/orchestrator? Or anything else else I've might have forgotten to consider?

FYI: @hmusum @mpolden

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/ZoneAppMigrationTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        proxyHostSpecs = prepareSystemApplication(zoneApp, NodeType.proxy, "routing");
+        nodeAdminHostSpecs = prepareSystemApplication(zoneApp, NodeType.host, "node-admin");
+        zoneAppHostSpecs = concat(proxyHostSpecs, nodeAdminHostSpecs, Collectors.toList());
+        tester.activate(zoneApp, zoneAppHostSpecs);
+
+        assertEquals(1, nodeAdminHostSpecs.size()); // The newly provisioned host is prepared
+        newHost = tester.nodeRepository().getNode(newHost.hostname()).orElseThrow(); // Update newHost after it has been allocated
+        Set<Node> allNodesWithNewHost = concat(allNodes, Set.of(newHost), Collectors.toSet());
+        assertEquals(allNodesWithNewHost, Set.copyOf(tester.nodeRepository().getNodes()));
+        // The new host is allocated to zone-app, while the old ones are still allocated to tenant-host app
```
Comment (hakonhall) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/ZoneAppMigrationTest.java]: And these will be inactivated, dirtied, readied, and activated into the tenant-host at the next release w/o the type=host cluster?

**Review by hakonhall — APPROVED** [review]

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/ZoneAppMigrationTest.java (hunk 1 above)
Comment (freva) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/ZoneAppMigrationTest.java]: Yes
